### PR TITLE
Adding tf.Example and TFRecords tutorial.

### PR DESCRIPTION
**Pull request by raskutti**
Working with Josh to get the first version of this tutorial checked in. Please let me know if you have any suggestions, happy to make changes.

Direct Links: [GitHub](https://github.com/raskutti/docs/blob/master/site/en/tutorials/load_data/tf-records.ipynb), [Colab](https://colab.sandbox.google.com/github/raskutti/docs/blob/master/site/en/tutorials/load_data/tf-records.ipynb)

### CODE REVIEW

**Comment by lamberta** [comment]
Hi Kiran, thanks!

@random-forests Are there other notebooks for loading data? Can we move this to `tutorials/data/tf-records.ipynb` ?

**Comment by lamberta** [comment]
Ah, nevermind, just saw Mark's other notebook for loading data.
Just rename this to `tf-records.ipynb`, thanks

**Comment by raskutti** [comment]
Done :) Renamed to tf-records and also changed title

File: site/en/tutorials/load_data/tf-records.ipynb [hunk 1]
```diff
@@ -0,0 +1,973 @@
...
+        "  \n",
+        "  label = example.features.feature['label'].int64_list.value[0]\n",
+        "  \n",
+        "  # Use the label to determine which image is in each record.\n",
+        "  # The cat image has label 0, and the bridge image has label 1.\n",
+        "\n",
+        "  if label == 0:\n",
+        "    cat_image_string = example.features.feature['image_raw'].bytes_list.value[0]\n",
+        "  if label == 1:\n",
+        "    bridge_image_string = example.features.feature['image_raw'].bytes_list.value[0]"
```
Comment (karmel) [inline site/en/tutorials/load_data/tf-records.ipynb]: This would be cleaner if you created a dict of `images[label] = bytes`. That removes the need to explicitly check for label, and is more scalable in the case that a user has many labels. Below, you can just iterate over the dict/grab the keys directly as necessary.

File: site/en/tutorials/load_data/tf-records.ipynb [hunk 2]
```diff
@@ -0,0 +1,973 @@
...
+    {
+      "metadata": {
+        "id": "KzYvfFgtYE7x",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "!touch cat_in_snow_from_tfrecords.jpg\n",
+        "!touch williamsburg_bridge_from_tfrecords.jpg"
```
Comment (karmel) [inline site/en/tutorials/load_data/tf-records.ipynb]: Is this necessary? I thought open() would create the files for writing if they don't exist.

File: site/en/tutorials/load_data/tf-records.ipynb [hunk 3]
```diff
@@ -0,0 +1,973 @@
...
+        "feature = {\n",
+        "      'feature0': _int64_feature(example_observation[0]),\n",
+        "      'feature1': _bytes_feature(example_observation[1]),\n",
+        "      'feature2': _int64_feature(example_observation[2]),\n",
+        "      'feature3': _float_feature(example_observation[3]),\n",
+        "}\n",
+        "\n",
+        "# Create a Features message using tf.train.Example.\n",
+        "\n",
+        "tfExample = tf.train.Example(features=tf.train.Features(feature=feature))\n",
```
[the remark this answers is not in the document]
Comment (raskutti) [reply site/en/tutorials/load_data/tf-records.ipynb]: Done, everywhere

File: site/en/tutorials/load_data/tf-records.ipynb [hunk 4]
```diff
@@ -0,0 +1,973 @@
...
+    {
+      "metadata": {
+        "id": "pX_jeiWDmLUD",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def write_example(features, writer):\n",
+        "  '''\n",
```
[the remark this answers is not in the document]
Comment (raskutti) [reply site/en/tutorials/load_data/tf-records.ipynb]: Done.

File: site/en/tutorials/load_data/tf-records.ipynb [hunk 5]
```diff
@@ -0,0 +1,973 @@
...
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "# Write the tf.Example observations to test.tfrecords.\n",
+        "\n",
+        "writer = tf.python_io.TFRecordWriter('test.tfrecords')\n",
+        "\n",
+        "for i in range(n_observations):\n",
+        "  write_example([feature0[i], feature1[i], feature2[i], feature3[i]], writer)\n",
```
[the remark this answers is not in the document]
Comment (raskutti) [reply site/en/tutorials/load_data/tf-records.ipynb]: Done.

File: site/en/tutorials/load_data/tf-records.ipynb [hunk 6]
```diff
@@ -0,0 +1,973 @@
...
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "y-Hjmee-fbLH",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Using The `Dataset` Structure"
```
[the remark this answers is not in the document]
Comment (raskutti) [reply site/en/tutorials/load_data/tf-records.ipynb]: Done, renamed to Dataset object

File: site/en/tutorials/load_data/tf-records.ipynb [hunk 7]
```diff
@@ -0,0 +1,973 @@
...
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "n_observations_to_print = 10\n",
+        "\n",
+        "counter = 0\n",
+        "\n",
+        "for record in dataset:\n",
```
[the remark this answers is not in the document]
Comment (raskutti) [reply site/en/tutorials/load_data/tf-records.ipynb]: This is cool! Thanks

File: site/en/tutorials/load_data/tf-records.ipynb (hunk 1 above)
Comment (raskutti) [reply site/en/tutorials/load_data/tf-records.ipynb]: Done

File: site/en/tutorials/load_data/tf-records.ipynb (hunk 2 above)
Comment (raskutti) [reply site/en/tutorials/load_data/tf-records.ipynb]: Done
